pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -354,67 +346,49 @@ type reconstructedVolume struct {
 	blockVolumeMapper   volumepkg.BlockVolumeMapper
 }
 
-// syncStates scans the volume directories under the given pod directory.
-// If the volume is not in desired state of world, this function will reconstruct
-// the volume related information and put it in both the actual and desired state of worlds.
-// For some volume plugins that cannot support reconstruction, it will clean up the existing
-// mount points since the volume is no long needed (removed from desired state)
-func (rc *reconciler) syncStates() {
+// reconstructVolumes tries to reconstruct the actual state of world by scanning all pods' volume
+// directories from the disk. For the volumes that cannot support or fail reconstruction, it will
+// put the volumes to possiblyOrphanVolumes to be cleaned up later when DesiredStateOfWorld
+// is populated.
+func (rc *reconciler) reconstructVolumes() {
+	defer rc.updateLastSyncTime()
 	// Get volumes information by reading the pod's directory
 	podVolumes, err := getVolumesFromPodDir(rc.kubeletPodsDir)
 	if err != nil {
 		klog.ErrorS(err, "Cannot get volumes from disk, skip sync states for volume reconstruction")
 		return
 	}
-	volumesNeedUpdate := make(map[v1.UniqueVolumeName]*reconstructedVolume)
-	volumeNeedReport := []v1.UniqueVolumeName{}
+	reconstructedVolumes := make(map[v1.UniqueVolumeName]*reconstructedVolume)
+	volumesInNodeStatus := []v1.UniqueVolumeName{}
```

Comment:
after the change above, reconstruct is performed first, before any reconcile operations, then actual state should be empty, so no need for this check?